### ISSUE TITLE
Fetch Feed and server pull not working

### ISSUE THREAD

**Issue by shanto12** [MISP/MISP#4437]
Openphish feed fetch and another server pull working after the MISP upgrade. the configuration is as follows.

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | ubuntu
| OS version (client)       | 16.04.4
| MISP version / git hash   | 2.4.103

### Expected behavior
New IOCs should be fetched and added to the fixed event or added as new events.
### Actual behavior
The job gets done but other than that no IOCs added.
### Steps to reproduce the behavior
Click on fetch feed for openphish or do a server pull
### Logs, screenshots, configuration dump, ...
![image](https://user-images.githubusercontent.com/1020247/55776631-bc4f5280-5ae0-11e9-8502-4f1e6a9fa5bc.png)

![image](https://user-images.githubusercontent.com/1020247/55776647-c6715100-5ae0-11e9-97ef-1759883469d0.png)

![image](https://user-images.githubusercontent.com/1020247/56004263-b1c5d080-5d0d-11e9-9f98-2af79006252c.png)

**Comment by shanto12** [comment]
I keep seeing the below warning in the file at location: "/var/www/MISP/app/tmp/logs/error.log"

2019-04-09 15:53:16 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 882]

**Comment by SteveClement** [comment]
Dear @shanto12 can you first please update your MISP to v2.4.103 and then try again.

**Comment by seldem** [comment]
I am at v2.4.105 and getting the same warning in error.log
trying to fetch data from Phishtank.
2019-04-23 17:11:07 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

**Comment by dharshanduck** [comment]
Yes this i can confirm, i have the same issue spread across a couple of feeds. 

![image](https://user-images.githubusercontent.com/9196526/56707403-3e7b8000-674b-11e9-812e-889398cff0dc.png)


```
2019-04-24 22:00:19 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:00:19 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:00:30 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:00:41 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:00:48 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:00:50 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:00:51 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:00:59 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:01:21 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:01:23 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:01:25 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:27:00 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:27:07 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:27:30 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:27:33 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:27:35 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:27:37 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:27:38 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-24 22:31:23 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:31:24 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 22:31:25 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 23:01:01 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-24 23:01:01 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:00:10 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:00:11 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:00:22 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:00:22 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:00:34 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:00:41 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:00:44 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:00:47 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:00:51 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:01:10 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:01:12 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:01:14 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:38:38 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:38:44 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:39:12 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:39:17 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:39:18 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:39:20 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:39:21 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-04-25 10:43:10 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:43:12 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-04-25 10:43:13 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
```
This happens only during a scheduled feed fetch
![image](https://user-images.githubusercontent.com/9196526/56707221-8d74e580-674a-11e9-8b1a-49a87f694d6a.png)


However, for my instance at least when its manual, `Fetch and store all feed data` button, all feeds fetch with out a hickup.

There is an additional issue with the caching of feeds where i receive an error. But its not related to the issue title here, unless @SteveClement would you like to see it?

**Comment by shanto12** [comment]
it worked for sometime after we downgraded our PHP. But when i changed the event ID of a feed to another event, it stopped working again with the same error.

> Dear @shanto12 can you first please update your MISP to v2.4.103 and then try again.

Its already in that version.

**Comment by joeinfo35** [comment]
Hi,
I am using v2.4.105 and I have a same problem.
2019-05-06 05:59:03 Warning: array_merge(): Argument #2 is not an array in [/var                                       /www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-05-06 05:59:04 Warning: Invalid argument supplied for foreach() in [/var/ww                                       w/MISP/app/Model/Feed.php, line 883]
2019-05-06 05:59:26 Warning: array_merge(): Argument #2 is not an array in [/var                                       /www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-05-06 05:59:26 Warning: Invalid argument supplied for foreach() in [/var/ww                                       w/MISP/app/Model/Feed.php, line 1121]
2019-05-06 11:59:03 Warning: array_merge(): Argument #2 is not an array in [/var                                       /www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-05-06 11:59:05 Warning: Invalid argument supplied for foreach() in [/var/ww                                       w/MISP/app/Model/Feed.php, line 883]
2019-05-06 11:59:26 Warning: array_merge(): Argument #2 is not an array in [/var                                       /www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-05-06 11:59:27 Warning: Invalid argument supplied for foreach() in [/var/ww                                       w/MISP/app/Model/Feed.php, line 1121]

**Comment by iglocska** [comment]
Currently reworking the feed system completely, stay tuned!

**Comment by kalyparker** [comment]
Hi,
Do you have any news on this behavior?
On misp v2.4.107 (af1c2c86d57438ee0ea3c3946486719fa116d165) 
I still have the problem, but fun fact, it works half the time :)

For information the output:
```sudo -u www-data /var/www/MISP/app/Console/cake Server fetchFeed 1 69

Welcome to CakePHP v2.10.16 Console
---------------------------------------------------------------
App : app
Path:/var/www/MISP/app/
---------------------------------------------------------------
Warning Error: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]

2019-05-28 16:01:29 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
Job done.
```

**Comment by SteveClement** [comment]
Not sure, for the initial issue by @shanto12 I found that sometimes when the source feed is in a bad format, it gives me that issue.

But as @iglocska mentioned, there are some changes coming.

**Comment by rohit725** [comment]
I have same issue for the feeds with format other than misp i.e only two of them are working(CIRCL OSINT, The Botvrij.eu).
### Log
**resque-worker-error.log**
```
2019-06-06 08:00:04 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Warning Error: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]

2019-06-06 09:39:05 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 09:39:06 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Warning Error: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]

2019-06-06 09:39:26 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 09:39:26 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Job done.
Warning Error: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]

2019-06-06 09:56:15 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
Job failed.
Warning Error: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]

2019-06-06 09:58:31 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 09:58:32 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 10:07:48 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 10:09:34 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 10:12:24 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.
Warning Error: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

2019-06-06 10:18:41 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
Job done.

```

**Comment by dmolina213** [comment]
I am having the same issue when i fetch events on a single feed: I am running MISP on RHEL7

Invalid argument supplied for foreach() in [/var/ww
w/MISP/app/Model/Feed.php, line 883]

**Comment by EdAtt** [comment]
I'm seeing the same thing for multiple feeds.
Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]

Any idea when a fix will be available?

**Comment by dharshanduck** [comment]
@EdAtt as @iglocska mentioned 
>Currently reworking the feed system completely, stay tuned!

we'll have to wait

**Comment by ga2020tek** [comment]
Same issues here.

Version:  MISP 2.4.110
OS: Ubuntu Server 18.04

Currently noticed only for 3 feeds (using default ones included in MISP)

11 http://dns-bh.sagadc.org 
15 pan-unit42 
17 home.nuug.no 

2019-07-11 09:00:24 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-07-11 09:00:35 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]
2019-07-11 09:00:37 Warning: array_merge(): Argument #2 is not an array in [/var/www/MISP/app/Lib/cakephp/lib/Cake/Network/Http/HttpSocket.php, line 457]
2019-07-11 09:00:38 Warning: Invalid argument supplied for foreach() in [/var/www/MISP/app/Model/Feed.php, line 883]


Jobs Status:

Id | Date created | Date modified | Process ID | Worker | Job type | Input | Message | Organisation name | Status | Retries | Progress
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | 2019-07-09 15:53:09 | 2019-07-09 15:53:09 | 2bec905aca77c15a20de3d65959d22a3 | prio | publish_event | Event ID: 9 | Publishing. | ORGNAME | Unknown | 0 | Queued
71 | 2019-07-10 00:28:23 | 2019-07-10 00:28:23 | Part of scheduled feed fetch | default | feed_fetch | Feed: 11 | Saving data. | ORGNAME | Unknown | 0 | 50%
76 | 2019-07-10 00:28:26 | 2019-07-10 00:28:26 | Part of scheduled feed fetch | default | feed_fetch | Feed: 15 | Saving data. | ORGNAME | Unknown | 0 | 50%
79 | 2019-07-10 00:28:27 | 2019-07-10 00:28:27 | Part of scheduled feed fetch | default | feed_fetch | Feed: 17 | Saving data. | ORGNAME | Unknown | 0 | 50%
130 | 2019-07-10 09:00:56 | 2019-07-10 09:00:56 | Part of scheduled feed fetch | default | feed_fetch | Feed: 11 | Saving data. | ORGNAME | Unknown | 0 | 50%
137 | 2019-07-10 09:01:03 | 2019-07-10 09:01:03 | Part of scheduled feed fetch | default | feed_fetch | Feed: 17 | Saving data. | ORGNAME | Unknown | 0 | 50%
188 | 2019-07-11 09:00:38 | 2019-07-11 09:00:38 | Part of scheduled feed fetch | default | feed_fetch | Feed: 11 | Saving data. | ORGNAME | Unknown | 0 | 50%
195 | 2019-07-11 09:00:40 | 2019-07-11 09:00:40 | Part of scheduled feed fetch | default | feed_fetch | Feed: 17 | Saving data. | ORGNAME | Unknown | 0 |

**Comment by JakubOnderka** [comment]
This should be fixed in #5052. Or at least error logs will provide more information what happen.
